File: FriTD/src/Entities/Effect.java
package Entities;

public class Effect {

	private int remainingTurns;
	private int dmg;
	private double slow;
	
	public Effect(int remainingTurns, int dmg, double slow) {
		super();
		this.remainingTurns = remainingTurns;
		this.dmg = dmg;
		this.slow = slow;
	}

	public int getRemainingTurns() {
		return remainingTurns;
	}

	public int getDmg() {
		return dmg;
	}

	public double getSlow() {
		return slow;
	}

	public void reduceTTL() {
		remainingTurns--;
	}
	
	
	
	
}

File: FriTD/TD/Entities/Effect.cs
using TD.Enums;

namespace TD.Entities
{
    public class Effect
    {

        public int RemainingTurns { get; set; }
        public double PrimaryDmg { get; set; }
        public DmgType DmgType { get; set; }
        public double SecondaryDmg { get; set; }
        public double Slow { get; set; }
        public bool PrimaryDmgUsed { get; set; }
        public int Splash { get; set; }
        public int SplashRadius { get; set; }

        public double Dmg(int armor, int magicResist)
        {
            //TODO splash here
            var dmg = PrimaryDmgUsed ? SecondaryDmg : PrimaryDmg;
            PrimaryDmgUsed = true;
            var decrease = DmgType == DmgType.Magical ? magicResist : armor;
            return dmg*((100 - decrease)*1.0/100);
        }

        public void ReduceTtl()
        {
            RemainingTurns--;
        }

       
    }
}
